library.c

```diff
@@ -1047,10 +1047,11 @@ PHP_REDIS_API void redis_parse_client_list_response(char *response, zval *z_resu
                     /* Add as a long or string, depending */
                     if(is_numeric == 1) {
                         add_assoc_long(&z_sub_result, key, atol(value));
-                        efree(value);
                     } else {
                         add_assoc_string(&z_sub_result, key, value);
                     }
+                    efree(value); // Either way, it doesn't use the original string.
```

Comment:
If you want to avoid duplicating strings unnecessarily, use the zend_strings instead of char* : See zend_hash_add: https://php-lxr.adamharvey.name/source/xref/PHP-7.0/Zend/zend_hash.h#80
(May be missing a layer of pointers or pointer dereferencing in this example)

```
HashTable*ht = ZVAL_ARR(z_subresult); // outside result
zend_hash_add(ht, key, value);
```
(This may involve incrementing ref counts if used multiple times, or it might be done automatically if used as array keys, I'm not sure)